src/main/java/com/ericsson/ei/handlers/StatusHandler.java

```diff
@@ -0,0 +1,117 @@
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package com.ericsson.ei.handlers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import com.ericsson.ei.listeners.RMQConnectionListener;
+import com.ericsson.ei.status.Status;
+import com.ericsson.ei.status.StatusData;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * This class updates the statuses of EI, RabbitMQ and MongoDB on a given timer interval. When
+ * requested it returns status as a JsonNode.
+ *
+ */
+@Component
+public class StatusHandler {
+
+    private static final String INITIAL_DELAY_OF_FIRST_STATUS_UPDATE = "1000";
+    private static final String INTERVAL_TO_RUN_STATUS_UPDATES = "30000";
+
+    private StatusData statusData = new StatusData();
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Autowired
+    private MongoDBHandler mongoDBHandler;
+
+    @Autowired
+    private RMQConnectionListener rmqConnectionListener = new RMQConnectionListener();
+
+    /**
+     * Scheduled method to run status update on a given interval.
+     */
+    @Scheduled(initialDelayString = INITIAL_DELAY_OF_FIRST_STATUS_UPDATE, fixedRateString = INTERVAL_TO_RUN_STATUS_UPDATES)
+    public void run() {
+        updateCurrentStatus();
+    }
+
+    /**
+     * This method returns a JsonNode containing all fields and values in the statusData
+     *
+     * @return statusData
+     */
+    public JsonNode getCurrentStatus() {
+        return objectMapper.valueToTree(statusData);
+    }
+
+    private void updateCurrentStatus() {
+        Status mongoDBStatus = getMongoDBStatus();
+        statusData.setMongoDBStatus(mongoDBStatus);
+
+        Status rabbitMQStatus = getRabbitMQStatus();
+        statusData.setRabbitMQStatus(rabbitMQStatus);
+
+        Status eiffelIntelligenceStatus = getEiffelIntelligenceStatus();
+        statusData.setEiffelIntelligenceStatus(eiffelIntelligenceStatus);
+    }
+
+    private Status getMongoDBStatus() {
+        Status status = Status.UNAVAILABLE;
+
+        if (mongoDBHandler.isMongoDBServerUp()) {
+            status = Status.AVAILABLE;
+        }
+
+        return status;
+    }
+
+    private Status getRabbitMQStatus() {
+        Status status = Status.UNAVAILABLE;
+
+        if (rmqConnectionListener.isConnected()) {
+            status = Status.AVAILABLE;
+        }
+
+        return status;
+    }
+
+    private Status getEiffelIntelligenceStatus() {
+        Status status = Status.UNAVAILABLE;
+        boolean availability = getServiceAvailability();
+
+        if (availability) {
+            status = Status.AVAILABLE;
+        }
+
+        return status;
+    }
+
+    private boolean getServiceAvailability() {
+        boolean availability;
```

Comment:
Do we really need this EiffelIntelligenceStatus(that includes RabbitMq and Mongodb) when MongoDb and RabbitMq status is already present in the status result back to EI Frotnend WebUI. They say same thing.
If you add a ER check later, then it will be a separate Status field in the result. If I understand the code correct here.
So, is the EiffelIntelligence Status really needed? Seem to be same result.

Reply:
Since it do reflect the EI status yes.

Reply:
But they contain same thing,, don't see why to duplicate info in the result.